core/domain/user_services_test.py

```diff
@@ -129,3 +133,236 @@ def test_get_user_id_from_username(self):
+            self.assertIsNotNone(user_services.get_user_settings(owner_id)
+            .first_contribution_msec)
+        for editor_id in exploration_rights.editor_ids:
+            self.assertIsNotNone(user_services.get_user_settings(editor_id)
+            .first_contribution_msec)
+
+    def test_contribution_msec_does_not_change_if_exploration_unpublished(self):
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_EMAIL])
+
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.login(self.OWNER_EMAIL)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.owner_id, end_state_name='End')
+        self.init_state_name = exploration.init_state_name
+        rights_manager.publish_exploration(self.owner_id, self.EXP_ID)
+        rights_manager.unpublish_exploration(self.admin_id, self.EXP_ID)
+
+        # Test that contribution time is not eliminated if exploration is
+        # unpublished.
+        self.assertIsNotNone(user_services.get_user_settings(self.owner_id)
+            .first_contribution_msec)
+
+    def test_contribution_msec_updates_on_published_collections(self):
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.login(self.ADMIN_EMAIL)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+
+        collection = self.save_new_valid_collection(
+            self.COL_ID, self.admin_id, title=self.COLLECTION_TITLE,
+            category=self.COLLECTION_CATEGORY,
+            objective=self.COLLECTION_OBJECTIVE,
+            exploration_id=self.EXP_ID)
+
+        rights_manager.publish_collection(self.admin_id, self.COL_ID)
+
+        # Test all owners and editors of collection after publication have
+        # updated first contribution times.
+        self.assertIsNotNone(user_services.get_user_settings(self.admin_id)
+            .first_contribution_msec)
+
+        # Test editor of published collection has updated first contribution time.
+        rights_manager.release_ownership_of_collection(self.admin_id,
+            self.COL_ID)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        collection_services.update_collection(
+            self.editor_id, self.COL_ID, [{
+                'cmd': 'edit_collection_property',
+                'property_name': 'title',
+                'new_value': 'Some new title'
+            }], 'Changed the title')
+
+        self.assertIsNotNone(user_services.get_user_settings(self.editor_id)
+            .first_contribution_msec)
+
+    def test_contribution_msec_updates_on_unpublished_collections(self):
```

Comment:
should this be a test that it does not update? I.e.

    test_contribution_msec_does_not_update_until_collection_is_published

Ditto above.